argilla-server/src/argilla_server/contexts/datasets.py

```diff
@@ -936,7 +936,9 @@ async def create_response(
         )
 
         await db.flush([response])
+        await distribution.refresh_record_status(db, record, autocommit=False)
         await _touch_dataset_last_activity_at(db, record.dataset)
+        await search_engine.index_records(record.dataset, [record])
         await search_engine.update_record_response(response)
```

Comment:
We need to check if only the record index is required and the one for response is not necessary.